### PR TITLE
Emit json when radiff2 is run with -Cj ##diff

### PR DESCRIPTION
**Pull request by erfur**
- [x] Mark this if you consider it ready to merge
- [ ] I've added tests (optional)
- [ ] I wrote some lines in the [book](https://github.com/radareorg/radare2book) (optional)

**Description**

Radiff2 when run with `-AAC` creates a table of similar functions based on their similarity scores. This information is not available as json, so I added a new function `r_core_diff_show_json`. The output of this function consists of an array of objects that have the following template:

```
    {
        "addr": 4203120,
        "name": "entry0",
        "size": 88,
        "diff_addr": 4203248,
        "diff_name": "entry0",
        "diff_size": 88,
        "match": "MATCH",
        "dist": 0.965909
    },
    {
        "addr": 4213228,
        "name": "fcn.004049ec",
        "size": 664,
        "match": "NEW",
        "dist": 0.000000
    },
```

### CODE REVIEW

File: libr/core/gdiff.c [hunk 1]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
+	RAnalFunction *f;
+	PJ *pj = pj_new ();
+
+	if (!pj) {
+		return;
+	}
+
+	fcns = r_anal_get_fcns (c->anal);
+	if (r_list_empty (fcns)) {
+		eprintf ("No functions found, try running with -A or load a project\n");
```
Comment (trufae) [inline libr/core/gdiff.c:177]: Use the new R_LOG_INFO() macro

**Review by trufae — CHANGES_REQUESTED** [review]

**Comment by trufae** [comment]
Thanks for the improvement!

File: libr/core/gdiff.c [hunk 2]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
 	}
 }
+
+/* Iterate available diffs and print json output */
+R_API void r_core_diff_show_json(RCore *c, RCore *c2) {
+	RList *fcns = r_anal_get_fcns (c->anal);
+	const char *match;
+	RListIter *iter;
+	RAnalFunction *f;
+	PJ *pj = pj_new ();
```
[the remark this answers is not in the document]
Comment (erfur) [reply libr/core/gdiff.c]: You mean r_core_pj_new right?

File: libr/core/gdiff.c (hunk 1 above)
Comment (erfur) [reply libr/core/gdiff.c]: Does it make sense to use `R_LOG_ERROR()` here?

File: libr/core/gdiff.c [hunk 3]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
+	const char *match;
+	RListIter *iter;
+	RAnalFunction *f;
+	PJ *pj = r_core_pj_new (c);
+
+	if (!pj) {
+		return;
+	}
+
+	fcns = r_anal_get_fcns (c->anal);
```
Comment (trufae) [inline libr/core/gdiff.c:175]: This var is assigned twice

File: libr/core/gdiff.c [hunk 4]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
+	if (r_list_empty (fcns)) {
+		R_LOG_INFO ("No functions found, try running with -A or load a project\n");
```
Comment (trufae) [inline libr/core/gdiff.c:177]:
```suggestion
		R_LOG_INFO ("No functions found, try running with -A or load a project");
```

**Review by trufae — CHANGES_REQUESTED** [review]

File: libr/core/gdiff.c (hunk 3 above)
Comment (erfur) [reply libr/core/gdiff.c:175]: The calls are different, first one's argument is `c->anal` while the second one is `c2->anal`.

File: libr/core/gdiff.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (trufae) [reply libr/core/gdiff.c]: Yes

File: libr/core/gdiff.c [hunk 5]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
+	RAnalFunction *f;
+	PJ *pj = r_core_pj_new (c);
+
+	if (!pj) {
+		return;
+	}
+
+	fcns = r_anal_get_fcns (c->anal);
+	if (r_list_empty (fcns)) {
+		R_LOG_INFO ("No functions found, try running with -A or load a project");
```
Comment (trufae) [inline libr/core/gdiff.c:177]:
```suggestion
		R_LOG_ERROR ("No functions found, try running with -A or load a project");
```

File: libr/core/gdiff.c [hunk 6]
```diff
@@ -159,3 +159,79 @@ R_API void r_core_diff_show(RCore *c, RCore *c2) {
...
+		return;
```
Comment (trufae) [inline libr/core/gdiff.c:178]: Yes, error seems more correct indeed.

**Review by trufae — APPROVED** [review]

**Comment by trufae** [comment]
Merged!

**Pull request merged**